libs/langchain/langchain/embeddings/open_clip.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Dict, List
+
+from langchain.pydantic_v1 import BaseModel, root_validator
+from langchain.schema.embeddings import Embeddings
+
+
+class OpenCLIPEmbeddings(BaseModel, Embeddings):
+    model: Any
+    preprocess: Any
+    tokenizer: Any
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that open_clip and torch libraries are installed."""
+        try:
+            import open_clip
+
+            model_name = "ViT-B-32"
+            checkpoint = "laion2b_s34b_b79k"
+            model, _, preprocess = open_clip.create_model_and_transforms(
+                model_name=model_name, pretrained=checkpoint
+            )
+            tokenizer = open_clip.get_tokenizer(model_name)
+            values["model"] = model
+            values["preprocess"] = preprocess
+            values["tokenizer"] = tokenizer
+
+        except ImportError:
+            raise ImportError(
+                "Please ensure both open_clip and torch libraries are installed. "
+                "pip install open_clip_torch torch"
+            )
+        return values
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        text_features = [
+            self.model.encode_text(self.tokenizer(text)).tolist() for text in texts
+        ]
+        return text_features
+
+    def embed_query(self, text: str) -> List[float]:
+        return self.embed_documents([text])[0]
+
+    def embed_image(self, images: List[float]) -> List[List[float]]:
```

Comment:
The type on `images` seems like it should be a list of something 2d instead of floats

Reply:
good pt, in notebook seems like we're passing in list of np array even?

Reply:
Yes, that's right. Image loading should go from compressed img file (`.JPEG`) -> decoded raw pixels w/ `PIL` -> `np.array`.